### PR TITLE
CSV export of duplicate items

### PR DESCRIPTION
**Pull request by poyuki**
# [TR-4908](https://oat-sa.atlassian.net/browse/TR-4908)

## Summary
The goals of change to separate duplicated items in exported csv results
4 different strategy for items result export provided: 
- item_entity
[results_export_delivery_of_test_5_asdasdad_i638f606c153a7182f8ea00bf85fc928e_2022120618570427.csv](https://github.com/oat-sa/extension-tao-outcomeui/files/10169041/results_export_delivery_of_test_5_asdasdad_i638f606c153a7182f8ea00bf85fc928e_2022120618570427.csv)

- item_instance_label
[results_export_delivery_of_test_5_asdasdad_i638f606c153a7182f8ea00bf85fc928e_2022120618560576.csv](https://github.com/oat-sa/extension-tao-outcomeui/files/10169038/results_export_delivery_of_test_5_asdasdad_i638f606c153a7182f8ea00bf85fc928e_2022120618560576.csv)

- item_instance_item_ref
[results_export_delivery_of_test_5_asdasdad_i638f606c153a7182f8ea00bf85fc928e_2022120618535894.csv](https://github.com/oat-sa/extension-tao-outcomeui/files/10169024/results_export_delivery_of_test_5_asdasdad_i638f606c153a7182f8ea00bf85fc928e_2022120618535894.csv)

- item_instance_label_item_ref
[results_export_delivery_of_test_5_asdasdad_i638f606c153a7182f8ea00bf85fc928e_2022120618522588.csv](https://github.com/oat-sa/extension-tao-outcomeui/files/10169013/results_export_delivery_of_test_5_asdasdad_i638f606c153a7182f8ea00bf85fc928e_2022120618522588.csv)

## How to test
- install branch
- create and execute delivery with multiple times used item
- export csv in result page
- apply different strategies

## TODO
- validate different corner cases of functionality (preview on `Export Table` page)
- validate possibility to adjust existed or create new tests
- adjust README

### CODE REVIEW

**Review by kilatib — COMMENTED** [review]
LGTM just needs clarification about the flag change and for code styles (mentioned in tests)

File: model/export/SingleDeliveryResultsExporter.php [hunk 1]
```diff
@@ -352,7 +349,7 @@ protected function getExporter(array $result)
      */
     protected function getExportData($exporter)
     {
-        return $exporter->export(true, false, ',', '"', false);
+        return $exporter->export(false, false, ',', '"', false);
```
[the remark this answers is not in the document]
Comment (poyuki) [reply model/export/SingleDeliveryResultsExporter.php:349]: sure this [flag](https://github.com/oat-sa/tao-core/blob/master/models/classes/export/implementation/CsvExporter.php#L55) used to push keys from data array as [first row of csv](https://github.com/oat-sa/tao-core/blob/master/models/classes/export/implementation/CsvExporter.php#L67-L69) i'm reworked a way how column names and data collected to provide possibility export duplicated labels for different instances of the same item and push [first row with column names](https://github.com/oat-sa/extension-tao-outcomeui/pull/450/files#diff-7b989e699f5aa7413a48cfa43cb305022912500930a14e56bfd3c2ee173f3ff9R322) explicitly

[the remark this answers is not in the document]
Comment (kilatib) [reply model/export/SingleDeliveryResultsExporter.php:362]: thank you for good description

[the remark this answers is not in the document]
Comment (kilatib) [reply model/export/SingleDeliveryResultsExporter.php:362]: If it not difficult could you please add cuple unittest for new way of getting kay in result array?

**Review by wazelin — APPROVED** [review]
Superb job, @poyuki 💪 

- [x] New code is covered by tests (if applicable) I guess that will be fixed soon
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by wazelin — APPROVED** [review]

**Review by kilatib — APPROVED** [review]
Thank you very much !! for tests LGTM
